service/src/main/java/uk/nhs/adaptors/gp2gp/common/task/TaskHandler.java

```diff
@@ -19,21 +21,49 @@
     private final TaskDefinitionFactory taskDefinitionFactory;
     private final TaskExecutorFactory taskExecutorFactory;
     private final MDCService mdcService;
+    private final ProcessFailureHandlingService processFailureHandlingService;
 
+    /**
+     * @return True if the message has been processed. Otherwise, false.
+     */
     @SuppressWarnings({"unchecked", "rawtypes"})
-    public void handle(Message message) {
-        String taskType = null;
-        String body = null;
+    @SneakyThrows
+    public boolean handle(Message message) {
+        TaskDefinition taskDefinition = null;
+
+        try {
+            taskDefinition = readTaskDefinition(message);
+
+            if (!processFailureHandlingService.hasProcessFailed(taskDefinition.getConversationId())) {
+                executeTask(taskDefinition);
+            } else {
+                LOGGER.warn(
```

Comment:
Shouldn't aborted task return false to not ack the message?

Reply:
At this point we know the process (conversation?) has failed and we're not going to perform any more tasks. The message can be considered processed, as we've done everything we were supposed to with it.

We should exclude "Send NACK" task from that, though, because NACK message should always be sent - I'll update the code.

Reply:
yes, looking at it form this perspective makes sense, thanks for clarifying.